update.ps1

```diff
@@ -1,179 +1,194 @@
+        Write-Verbose "Error at Line '$($ex.InvocationInfo.ScriptLineNumber)': $($ex.InvocationInfo.Line). Error: $($ex.Exception.Message)"
+        
+        $outputContext.AuditLogs.Add([PSCustomObject]@{
+                Action  = "CreateAccount" # Optionally specify a different action for this audit log
+                Message = $auditMessage
+                IsError = $false
+            })
+        throw "Error querying data from SQL Lite DB"
     }
 
-}
-function GenerateLastName {
-    [cmdletbinding()]
-    Param (
-        [object]$person
-    )
+    $previousAccount = [PSCustomObject]@{
+        achternaam     = $currentAccount.achternaam
+        tussenvoegsel  = $currentAccount.tussenvoegsel
+        voornaam       = $currentAccount.voornaam
+        gebruikersnaam = $currentAccount.gebruikersnaam
+        geslacht       = $currentAccount.gender
+        externalId     = $currentAccount.externalId
+        email          = $currentAccount.email
+    }
 
-    try {
-        $suffix = "";
-        $givenname = if ([string]::IsNullOrEmpty($person.Name.Nickname)) { $person.Name.Initials.Substring(0, 1) }else { $person.Name.Nickname }
-        $FamilyNamePrefix = $person.Name.FamilyNamePrefix
-        $FamilyName = $person.Name.FamilyName           
-        $PartnerNamePrefix = $person.Name.FamilyNamePartnerPrefix
-        $PartnerName = $person.Name.FamilyNamePartner 
-        $convention = $person.Name.Convention
-
-        $LastName = ""
-
-        switch ($convention) {
-            "B" {
-                #  $LastName += if (-NOT([string]::IsNullOrEmpty($FamilyNamePrefix))) { $FamilyNamePrefix + " " }
-                $LastName += $FamilyName  
-            }
-            "P" {
-                #   $LastName += if (-NOT([string]::IsNullOrEmpty($FamilyNamePrefix))) { $FamilyNamePrefix + " " }
-                $LastName += $PartnerName                    
-            }
-            "BP" {
-                #   $LastName += if (-NOT([string]::IsNullOrEmpty($FamilyNamePrefix))) { $FamilyNamePrefix + " " }
-                $LastName += $FamilyName + " - "
-                $LastName += if (-NOT([string]::IsNullOrEmpty($PartnerNamePrefix))) { $PartnerNamePrefix + " " }
-                $LastName += $PartnerName
+
+    # Calculate changes between current data and provided data
+    $splatCompareProperties = @{
+        ReferenceObject  = @($previousAccount.PSObject.Properties) # Only select the properties to update
+        DifferenceObject = @($account.PSObject.Properties) # Only select the properties to update
+    }
+    $changedProperties = $null
+    $changedProperties = (Compare-Object @splatCompareProperties -PassThru)
+    $oldProperties = $changedProperties.Where( { $_.SideIndicator -eq '<=' })
+    $newProperties = $changedProperties.Where( { $_.SideIndicator -eq '=>' })
+
+    if (($newProperties | Measure-Object).Count -ge 1) {
+
+        If (Test-Path $database) {   
```

Comment:
I think we can remove this if statement because, we've already queried the database on line 53. If the database didn't exist, we would have known. However, a check if it exists is justified. Although we might need to relocate this check to the beginning of our code.